### PR TITLE
Add Saleae Logic flatpak manifest

### PR DESCRIPTION
**Pull request by mripard**
The Saleae Logic application is a capture application to drive the Saleae
logic analyzers.

Signed-off-by: Maxime Ripard <maxime.ripard@anandra.org>

### CODE REVIEW

File: com.saleae.Logic.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        "--socket=x11",
+        "--share=ipc",
+        "--device=all",
+        "--env=LD_LIBRARY_PATH=/app/extra/lib",
+        "--env=QT_ACCESSIBILITY=0",
+        "--env=NO_AT_BRIDGE=1"
+    ],
+    "build-options": {
+        "build-args": [
+            "--with-appdir"
```
Comment (eszlari) [inline com.saleae.Logic.json]: What's this for?

**Comment by eszlari** [comment]
You will need to add such a file:
https://github.com/flathub/com.visualstudio.code/blob/master/flathub.json

File: com.saleae.Logic.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Logic</name>
+  <summary>Drive your Saleae Logic analyzer</summary>
+  <description>
+    <p>Use your Saleae Logic analyzer on Linux. Start captures, analyze them and save them.</p>
+  </description>
+  <categories>
+    <category>Development</category>
+  </categories>
+  <content_rating type="oars-1.0">
```
Comment (eszlari) [inline com.saleae.Logic.appdata.xml]: Use `<content_rating type="oars-1.1" />` and remove the rest.

File: com.saleae.Logic.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+<component>
+  <id type="desktop">com.saleae.Logic.desktop</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Logic</name>
+  <summary>Drive your Saleae Logic analyzer</summary>
+  <description>
+    <p>Use your Saleae Logic analyzer on Linux. Start captures, analyze them and save them.</p>
+  </description>
+  <categories>
+    <category>Development</category>
```
Comment (eszlari) [inline com.saleae.Logic.appdata.xml]: Remove categories here and add it to `com.saleae.Logic.desktop` instead.

File: com.saleae.Logic.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2018 Maxime Ripard <maxime.ripard@anandra.org> -->
+<component>
+  <id type="desktop">com.saleae.Logic.desktop</id>
```
Comment (eszlari) [inline com.saleae.Logic.appdata.xml]: `.desktop` can be dropped

File: com.saleae.Logic.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Logic</name>
+  <summary>Drive your Saleae Logic analyzer</summary>
+  <description>
+    <p>Use your Saleae Logic analyzer on Linux. Start captures, analyze them and save them.</p>
```
Comment (nedrichards) [inline com.saleae.Logic.appdata.xml]: This description should be a little fuller.

**Comment by nedrichards** [comment]
Appdata needs at least one screenshot.

**Comment by nedrichards** [comment]
bot, build com.saleae.Logic

File: com.saleae.Logic.json (hunk 1 above)
Comment (nedrichards) [reply com.saleae.Logic.json]: More importantly, it breaks the sandboxed build https://flathub.org/builds/#/builders/2/builds/3703

**Comment by mripard** [comment]
I've pushed an updated version some time ago that should address your comments (hopefully), but I haven't received any feedback since then (while on the first iteration it was very fast). Is there anything else I need to do?

**Comment by nedrichards** [comment]
bot, build com.saleae.Logic

**Comment by nedrichards** [comment]
@mripard nothing really, I want to test it locally myself (which you really need with extra-data apps) other than that the skeleton lgtm

**Comment by nedrichards** [comment]
I installed locally to test and the application doesn't run at all for me. Does it need hardware plugged in to show up? On first run I got:

```
nick@endless-nedr:~/Projects/mripard/flathub$ flatpak run com.saleae.Logic
Qt: Session management error: Could not open network socket
Root Path: /app/extra/bin  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 158 ]
working dir parameter: /var/data  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 169 ]
Using user selected writeable directory  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 175 ]
Writeable Root Path: /var/data  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 185 ]
Temp Folder path: /tmp  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 191 ]
Release Mode; analyzer path is /app/extra/bin/Analyzers  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 201 ]
Settings folder does not exist, creating it...  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 236 ]
Errors folder does not exist, creating it...  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 251 ]
Databases folder does not exist, creating it...  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 268 ]
Calibration folder does not exist, creating it...  [ /home/build/ob_logic/Source/Environment.cpp; Environment; 279 ]
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
qt.network.ssl: QSslSocket: cannot resolve CRYPTO_num_locks
qt.network.ssl: QSslSocket: cannot resolve CRYPTO_set_id_callback
qt.network.ssl: QSslSocket: cannot resolve CRYPTO_set_locking_callback
qt.network.ssl: QSslSocket: cannot resolve ERR_free_strings
qt.network.ssl: QSslSocket: cannot resolve EVP_CIPHER_CTX_cleanup
qt.network.ssl: QSslSocket: cannot resolve EVP_CIPHER_CTX_init
qt.network.ssl: QSslSocket: cannot resolve sk_new_null
qt.network.ssl: QSslSocket: cannot resolve sk_push
qt.network.ssl: QSslSocket: cannot resolve sk_free
qt.network.ssl: QSslSocket: cannot resolve sk_num
qt.network.ssl: QSslSocket: cannot resolve sk_pop_free
qt.network.ssl: QSslSocket: cannot resolve sk_value
qt.network.ssl: QSslSocket: cannot resolve SSL_library_init
qt.network.ssl: QSslSocket: cannot resolve SSL_load_error_strings
qt.network.ssl: QSslSocket: cannot resolve SSL_get_ex_new_index
qt.network.ssl: QSslSocket: cannot resolve SSLv2_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv3_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv23_client_method
qt.network.ssl: QSslSocket: cannot resolve SSLv2_server_method
qt.network.ssl: QSslSocket: cannot resolve SSLv3_server_method
qt.network.ssl: QSslSocket: cannot resolve SSLv23_server_method
qt.network.ssl: QSslSocket: cannot resolve X509_STORE_CTX_get_chain
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_noconf
qt.network.ssl: QSslSocket: cannot resolve OPENSSL_add_all_algorithms_conf
qt.network.ssl: QSslSocket: cannot resolve SSLeay
qt.network.ssl: QSslSocket: cannot resolve SSLeay_version
qt.network.ssl: QSslSocket: cannot call unresolved function CRYPTO_num_locks
qt.network.ssl: QSslSocket: cannot call unresolved function CRYPTO_set_id_callback
qt.network.ssl: QSslSocket: cannot call unresolved function CRYPTO_set_locking_callback
qt.network.ssl: QSslSocket: cannot call unresolved function SSL_library_init
qt.network.ssl: QSslSocket: cannot call unresolved function SSLv23_client_method
qt.network.ssl: QSslSocket: cannot call unresolved function sk_num
```

Subsequent runs didn't show the QSslSocket stuff, but still nothing happened.

**Comment by mripard** [comment]
Thanks for taking the time to test. It can definitely run without a device connected. I've experienced some issues related to the SSL library in the past (which was iirc related to the SONAME of the one installed vs the one it uses). I'll update the runtimes on my end and try to reproduce your issue.

**Comment by LeandroStanger** [comment]
bot, build com.saleae.Logic

**Comment by LeandroStanger** [comment]
bot, build com.saleae.Logic

File: com.saleae.Logic.json [hunk 6]
```diff
@@ -0,0 +1,83 @@
+{
+    "app-id": "com.saleae.Logic",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
```
Comment (LeandroStanger) [inline com.saleae.Logic.json:4]: Update runtime to 19.08

File: com.saleae.Logic.json [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+            "name": "Logic",
+            "buildsystem": "simple",
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "com.saleae.Logic.desktop"
+                },
+                {
+                    "type": "file",
+                    "path": "com.saleae.Logic.appdata.xml"
```
Comment (LeandroStanger) [inline com.saleae.Logic.json:26]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by bilelmoussaoui** [comment]
@LeandroStanger Please don't bump PR that are a year old without any response otherwise they won't be auto-closed.

**Comment by mripard** [comment]
Unfortunately I don't have access to the hardware that this is supposed to interact with, so I guess it's just better to close the PR until someone can work on that again.

**Pull request closed without merging**